cmd/collector/config.go

```diff
@@ -365,25 +354,94 @@ func (w *TailLog) Validate() error {
 			}
 		}
 		if !foundValidType {
-			return fmt.Errorf("tail-log.static-target.log-type %s is not a valid log type", v.LogType)
+			return fmt.Errorf("host-log.file-target.log-type %s is not a valid log type", v.LogType)
 		}
 
 		for _, parserName := range v.Parsers {
 			if !parser.IsValidParser(parserName) {
-				return fmt.Errorf("tail-log.static-target.parsers %s is not a valid parser", parserName)
+				return fmt.Errorf("host-log.file-target.parsers %s is not a valid parser", parserName)
 			}
 		}
 	}
 
+	for _, v := range w.JournalTargets {
+		if err := v.Validate(); err != nil {
+			return fmt.Errorf("host-log.journal-target.%w", err)
+		}
+	}
+
 	for _, v := range w.Transforms {
 		if !transforms.IsValidTransformType(v.Name) {
-			return fmt.Errorf("tail-log.transforms %s is not a valid transform", v.Name)
+			return fmt.Errorf("host-log.transforms %s is not a valid transform", v.Name)
 		}
 	}
 
 	return nil
 }
 
+type JournalTarget struct {
+	Matches  []string `toml:"matches" comment:"Matches for the journal reader based on journalctl MATCHES. To select a systemd unit, use the field _SYSTEMD_UNIT. (e.g. '_SYSTEMD_UNIT=avahi-daemon.service' for selecting logs from the avahi-daemon service.)"`
+	Database string   `toml:"database" comment:"Database to store logs in."`
+	Table    string   `toml:"table" comment:"Table to store logs in."`
+	Parsers  []string `toml:"parsers" comment:"Parsers to apply sequentially to the log line."`
+}
+
+func (j *JournalTarget) Validate() error {
+	if j.Database == "" {
+		return errors.New("database must be set")
+	}
+	if j.Table == "" {
+		return errors.New("table must be set")
+	}
+
+	haveFirstFilter := false
+	for _, filter := range j.Matches {
+		if filter == "" {
+			return errors.New("match must have a value")
+		}
+
+		if filter == "+" {
+			if !haveFirstFilter {
+				return errors.New("matches must not start with +")
+			}
+			continue
+		}
+
+		splitVals := strings.Split(filter, "=")
+		if len(splitVals) != 2 {
+			return fmt.Errorf("match %s must be in the format key=value", filter)
+		}
+		if splitVals[0] == "" {
+			return fmt.Errorf("match %s must have a key", filter)
+		}
+		if splitVals[1] == "" {
+			return fmt.Errorf("match %s must have a value", filter)
+		}
+		haveFirstFilter = true
+	}
+
+	for _, parserName := range j.Parsers {
+		if !parser.IsValidParser(parserName) {
+			return fmt.Errorf("parsers %s is not a valid parser", parserName)
+		}
+	}
+
+	return nil
```

Comment:
Since `database` and `table` are used as part of the WAL filename, should we do some sort of validation and sanitization to these parameters to prevent any sort of path traversal vulnerabilities?

Reply:
This is a good question. I want to see how we're doing this over time.